### DISCUSSION TITLE
Partition occurs

### DISCUSSION THREAD

**Discussion by bahientn9x** [emqx/emqx#12782] [Q&A]
My cluster consists of 4 nodes. My cluster generated a partition error for the first time. I checked the log and at node 1 it said there was no response from node 2 and node 3 but all nodes had no reboot history. But all clients lost connection at the time of the incident. Why does one node failure cause the entire cluster to fail and cause all clients to lose connection? What could be the cause of this error? Thanks.

**Comment by bahientn9x** [comment]
Log node 1
![image](https://github.com/emqx/emqx/assets/37149979/9b10de4b-b59f-4663-94a1-3bce05839989)
Log node 2
![image](https://github.com/emqx/emqx/assets/37149979/3a295eb0-625b-4f06-8b9c-78eb8411b34b)
Log node 3
![image](https://github.com/emqx/emqx/assets/37149979/6bd737f8-42d1-4a89-9725-8c60130527f6)
Log node 4
2024-03-26T03:14:43.381806+07:00 [warning] msg: Stopping mria, mfa: mria:stop/1(134), reason: heal

**Comment by ieQu1** [comment]
Hello,

It might be a consequence of this Erlang/OTP patch: https://github.com/erlang/otp/pull/5611 that prevents "overlapping partitions" (by aggressively disconnecting nodes from each other), and its interaction with old the EMQX leader election algorithm that predates this behavior. 
We've had some discussions internally whether we want to disable the new behavior, or err on the side of caution, and chose the latter. 

Apparently it causes problems in real life, and given that starting from v5.7 EMQX will use Raft protocol to replicate some of its internal state machines, we may as well start using it for Mria partition healing, and eventually phase out the parts of code that rely on the overlapping partition prevention.

Could you create a bug report for easier tracking?

**Comment by zmstone** [comment]
Hi @bahientn9x 
Which EMQX version is this ?

**Reply by bahientn9x** [reply]
version is 5.3.1
